### PR TITLE
[numpy] Fix ffi split 

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
#18132 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by Yiyan66** [comment]
@yzhliu

**Comment by sxjscience** [comment]
@Yiyan66 We need to add test

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -3901,6 +3901,8 @@ def split(ary, indices_or_sections, axis=0):
         If `indices_or_sections` is given as an integer, but
         a split does not result in equal division.
     """
+    while axis < 0:
+        axis += len(ary.shape)
```
Comment (sxjscience) [inline python/mxnet/ndarray/numpy/_op.py:3905]: This will fail in the symbolic mode. Do we also test in the symbolic mode?

**Comment by Yiyan66** [comment]
@mxnet-bot run ci [unix-gpu]

**Pull request merged**